btree/btree.c

```diff
@@ -2422,21 +2360,18 @@ static bool ff_invariant(const struct nd *node)
 {
 	const struct ff_head *h = ff_data(node);
 
-	return  _0C(h->ff_shift == segaddr_shift(&node->n_addr)) &&
-		_0C(node->n_skip_rec_count_check ||
-		    ergo(h->ff_level > 0, h->ff_used > 0));
+	return  _0C(h->ff_shift == segaddr_shift(&node->n_addr));
```

Comment:
Why the check is removed ?

Reply:
Initially, it was removed to accommodate segaddr since we were not using node descriptor. Shashank suggested that this ergo check is not useful. Hence, after reversing the segaddr changes, I didnt update this function.

Reply:
We need the check till we have efficient invariant function. 
@RadhaGulhane13 Please confirm if this check is needed or not?

Reply:
I have added this check to ignore this check : h->ff_level > 0, h->ff_used > 0
If @ydb242  is removing this check, h->ff_level > 0, h->ff_used > 0, no need to check the flag. 
But on the other hand, if we make this change, invariant does not seems to be inefficient. So, the other option is if you want to remove flag check , you can remove h->ff_level>0 check but keep h->ff_used>=0 check as it is .
But I think we need the suggestion from @shashank-parulekar  as well .

Reply:
@RadhaGulhane13 https://github.com/Seagate/cortx-motr/pull/819#discussion_r664072942

Reply:
okay .